cmd/provenanced/cmd/root.go

```diff
@@ -60,7 +63,25 @@ const (
 // NewRootCmd creates a new root command for provenanced. It is called once in the main function.
 // Providing sealConfig = false is only for unit tests that want to run multiple commands.
 func NewRootCmd(sealConfig bool) (*cobra.Command, params.EncodingConfig) {
+	// tempDir creates a temporary home directory.
+	var tempDir = func() string {
+		dir, err := os.MkdirTemp("", "provenanced")
+		if err != nil {
+			panic("failed to create temp dir: " + err.Error())
+		}
+		defer os.RemoveAll(dir)
+
+		return dir
+	}
```

Comment:
```suggestion
	tempDir, err := os.MkdirTemp("", "provenanced")
	if err != nil {
		panic(fmt.Errorf("failed to create temp dir: %w", err))
	}
	os.RemoveAll(tempDir)
```

Then below, when creating `tempApp`, provide `tempDir` instead of `tempDir()`.